### ISSUE TITLE
Missing Postgres types

### ISSUE THREAD

**Issue by Bilge** [cakephp/phinx#1831]
The Postgres adapter supports some DBMS specific types, such as `interval`, but not all. For example, `timestamp with time zone` and its shorthand, `timestamptz` are missing.

**Comment by dereuromark** [comment]
Do you want to make a PR to add them?
The prio so far was usually the ones that are agnostic across the DBs, but that doesn't mean we couldn't add also more specific ones.

**Comment by Bilge** [comment]
No, I completely gave up using the API methods. All migrations are just `execute()` statements in `up()` and `down()` configurations. Phinx will never support the full feature set of Postgres, and simply adding a few types doesn't even begin to bring it closer to that lofty goal. Moreover, there are some highly questionable defaults when using the API, such as assuming everything should have an autonumber column, which is actually an anti-pattern of database design. I don't need to go on, I'm sure you get my point.
